extensions/wikibase/src/org/openrefine/wikibase/qa/EditInspector.java

```diff
@@ -220,8 +221,15 @@ private void resolveWarningPropertyLabels() throws ExecutionException {
         if (entityCache != null) {
             List<EntityIdValue> propertyIdValues = warningStore.getWarnings().stream()
                     .flatMap(warning -> warning.getProperties().entrySet().stream()
-                            .filter(entry -> entry.getValue() instanceof PropertyIdValue)
-                            .map(entry -> (PropertyIdValue) entry.getValue()))
+                            .filter(entry -> entry.getValue() instanceof EntityIdValue)
+                            .filter(entry -> {
+                                if (entry instanceof SuggestedEntityIdValue) {
+                                    String label = ((SuggestedEntityIdValue) entry.getValue()).getLabel();
+                                    return label != null && !label.isEmpty();
```

Comment:
Nice! But I am confused by this `return` statement. Shouldn't it be the negation instead? Because we only want to fetch the label if it is not present yet, right? Looking at the new files in the `data/entitycache` directory, your code is likely fetching more entities than it needs to.
```suggestion
                                    return label == null || label.isEmpty();
```

Reply:
@wetneb Thanks for catching this; i missed updating this.